### PR TITLE
Rewards Greaselion - Github

### PR DESCRIPTION
**Pull request by emerick**
Resolves https://github.com/brave/brave-browser/issues/11463
Resolves https://github.com/brave/brave-browser/issues/11766
Resolves https://github.com/brave/brave-browser/issues/10434

Associated Greaselion script PR: https://github.com/brave/brave-site-specific-scripts/pull/20

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [x] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`, `npm run gn_check`)
- [x] Ran `git rebase master` (if needed).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Requested a security/privacy review as needed.
- [x] Added appropriate QA labels (`QA/Yes` or `QA/No`) to the associated issue
- [x] Added appropriate release note labels (`release-notes/include` or `release-notes/exclude`) to the associated issue
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:

**Note**: In order to test this PR, you must manually install the Github Greaselion script at the moment. This involves pulling down the appropriate branch and running the following commands (the example is Windows specific, it will need to be modified slightly for Mac/Linux):

```
npm run build
cp -R dist/Greaselion.json /c/Users/$USERNAME/AppData/Local/BraveSoftware/Brave-Browser/User\ Data/afalakplffnnnlkncjhbmahjfjhmlkal/1.0.29/1/
cp -R dist/scripts /c/Users/$USERNAME/AppData/Local/BraveSoftware/Brave-Browser/User\ Data/afalakplffnnnlkncjhbmahjfjhmlkal/1.0.29/1/
```

#### Pull Request

- Clean profile
- Run browser
- Install Github Greaselion script as explained above
- Restart browser
- Visit a Github pull request URL (e.g., `https://github.com/brave/brave-core/pull/6743`)
- Verify that correct publisher information appears in Rewards panel
- Verify that tipping works as expected
- Verify that verified checkmark appears in Rewards icon for verified publishers
- Verify that tip icon is inserted in expected locations

#### Issue

- Clean profile
- Run browser
- Install Github Greaselion script as explained above
- Restart browser
- Visit a Github issue URL (e.g., `https://github.com/brave/brave-browser/issues/12013`)
- Verify that correct publisher information appears in Rewards panel
- Verify that tipping works as expected
- Verify that verified checkmark appears in Rewards icon for verified publishers
- Verify that tip icon is inserted in expected locations

#### Commits

- Clean profile
- Run browser
- Install Github Greaselion script as explained above
- Restart browser
- Visit a Github commits page URL (e.g., `https://github.com/brave/brave-browser/commits/master`)
- Verify that correct publisher information appears in Rewards panel
- Verify that tipping works as expected
- Verify that verified checkmark appears in Rewards icon for verified publishers
- Verify that tip icon is inserted in expected locations

#### Starred Repos

- Clean profile
- Run browser
- Install Github Greaselion script as explained above
- Restart browser
- Visit a Github starred repos URL (e.g., `https://github.com/emerick?tab=stars`)
- Verify that correct publisher information appears in Rewards panel
- Verify that tipping works as expected
- Verify that verified checkmark appears in Rewards icon for verified publishers
- Verify that tip icon is inserted in expected locations

#### Organization Members

- Clean profile
- Run browser
- Install Github Greaselion script as explained above
- Restart browser
- Visit a Github organization members URL (e.g., `https://github.com/orgs/brave/people`)
- Verify that correct publisher information appears in Rewards panel
- Verify that tipping works as expected
- Verify that verified checkmark appears in Rewards icon for verified publishers
- Verify that tip icon is inserted in expected locations

#### Team Members

- Clean profile
- Run browser
- Install Github Greaselion script as explained above
- Restart browser
- Visit a Github team members URL (e.g., `https://github.com/orgs/brave/teams/brave-core/members`)
- Verify that correct publisher information appears in Rewards panel
- Verify that tipping works as expected
- Verify that verified checkmark appears in Rewards icon for verified publishers
- Verify that tip icon is inserted in expected locations

#### Gist

- Clean profile
- Run browser
- Install Github Greaselion script as explained above
- Restart browser
- Visit a Github Gist URL (e.g., `https://gist.github.com/darkdh/85fed2542190510aaee0f3ad91091818`)
- Verify that correct publisher information appears in Rewards panel
- Verify that tipping works as expected
- Verify that verified checkmark appears in Rewards icon for verified publishers
- Verify that tip icon is inserted in expected locations

#### Excluded URL

- Clean profile
- Run browser
- Install Github Greaselion script as explained above
- Restart browser
- Visit an excluded Github URL (e.g., `https://github.com/about`)
- Verify that correct publisher information appears in Rewards panel (github.com)
- Verify that tipping works as expected

#### Activity

- Clean profile
- Run browser
- Install Github Greaselion script as explained above
- Restart browser
- Visit a Github URL (e.g., `https://github.com/emerick`)
- Verify that `activity_info` database table is updated as expected
- Verify that auto-contribute table is updated to reflect new activity

#### History

- Navigate to various Github URLs in the same tab
- Verify that appropriate publisher information appears in the panel
- Verify that back/forward navigation updates the panel with correct publisher info

#### Tab Activation

- Navigate to various Github URLs in multiple tabs
- Verify that appropriate publisher information appears in the panel
- Verify that switching between tabs updates the panel with correct publisher info

#### Tipping non-Github Publishers

- Verify that tipping works as expected on non-Github publishers (we only made changes to Github and only on desktop)
- Verify that the panel looks as expected on non-Github publishers

### Mobile

No changes were made to mobile platforms, so verify that Github navigation / tipping / Rewards panel continue to work as expected on those platforms.

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: browser/extensions/api/brave_rewards_api.cc [hunk 1]
```diff
@@ -394,6 +394,9 @@ void BraveRewardsTipUserFunction::ShowTipDialog() {
   media_meta_data_dict.SetStringKey("mediaType", params->media_type);
   media_meta_data_dict.SetStringKey("publisherKey", params->publisher_key);
   media_meta_data_dict.SetStringKey("publisherName", params->publisher_name);
+  media_meta_data_dict.SetStringKey(
```
Comment (NejcZdovc) [inline browser/extensions/api/brave_rewards_api.cc:397]: are both `publisherName` and `publisherScreenName` used?

File: components/brave_rewards/resources/tip/components/siteBanner.tsx [hunk 2]
```diff
@@ -134,7 +134,7 @@ class Banner extends React.Component<Props, State> {
     } else if (mediaMetaData.mediaType === 'reddit') {
       return `u/${mediaMetaData.userName}`
     } else if (mediaMetaData.mediaType === 'github') {
-      return `@${mediaMetaData.userName}`
+      return `@${mediaMetaData.publisherScreenName}`
```
Comment (NejcZdovc) [inline components/brave_rewards/resources/tip/components/siteBanner.tsx:137]: I see that in `tipMediaUser.tsx` we use `'@' + mediaMetaData.publisherName`, where here we use `mediaMetaData.publisherScreenName`. I would suggest that we do the same in both places and this way you don't need to introduce new parameter

**Review by NejcZdovc — CHANGES_REQUESTED** [review]
1) black bubble (probably translation is not working)
image

2) verified publisher don't get image https://github.com/jumde
image

3) regular site should have github.com https://github.com/team
image

4) `@` should be the same. Now one is `@yrliou` another is `@Jocelyn Liu`

image

File: components/brave_rewards/resources/tip/components/siteBanner.tsx (hunk 2 above)
Comment (emerick) [reply components/brave_rewards/resources/tip/components/siteBanner.tsx:137]: I think that's just a bug in `tipMediaUser.tsx` (which is leading to issue #4 that you noticed below). The issue is that the Github tip dialog wants to show both the screen name (e.g. `emerick`) and the publisher name (e.g., `Emerick Rogul`). For some reason, the Twitter integration only ever showed one name so we were good.

**Comment by emerick** [comment]
@NejcZdovc Regarding missing localizations, it seems that there's some unreliability with retrieving localizations from a content script via `chrome.i18n.getMessage`. See https://bugs.chromium.org/p/chromium/issues/detail?id=83549 for some details. Restarting the browser fixes it for me. I was thinking that I could remove the call to `chrome.i18n.getMessage` from the content script and just send a request to the background page for the given string. We only have two strings, so I think the cost would be negligible. WDYT?

**Comment by emerick** [comment]
All feedback should be addressed.

**Review by zenparsing — APPROVED** [review]

**Review by NejcZdovc — APPROVED** [review]

**Pull request merged**